astropy/io/ascii/ui.py

```diff
@@ -271,6 +272,16 @@ def _guess(table, read_kwargs, format, fast_reader):
             # user supplies delimiter="|" but the guess wants to try delimiter=" ",
             # so skip the guess entirely.
             continue
+
+        if guess_kwargs not in filtered_guess_kwargs:
```

Comment:
I wonder how thee repeats happen in the first case? I remember that I looked into the guessing a while ago when I changed the order in which formats are tried in guessing, and I definitely remember that this is no easy logic. I'm not oppsed to fixing it in ths way if that is the best solution, I just wonder if this points to some deeper problem or if you know why `basic` is in there multiple times.